cmd/spire-server/cli/run/run.go

```diff
@@ -157,57 +164,53 @@ func mergeConfigs(c *server.Config, fileConfig, cliConfig *RunConfig) error {
 	return mergeConfig(c, cliConfig)
 }
 
-func mergeConfig(orig *server.Config, cmd *RunConfig) error {
+func mergeConfig(orig *server.Config, cmd *runConfig) error {
 	// Parse server address
-	if cmd.BindAddress != "" {
-		ip := net.ParseIP(cmd.BindAddress)
+	if cmd.Server.BindAddress != "" {
```

Comment:
Here's the check I was referring to above. I guess a test for this would be good, but not required in this pr